tests/test_llm_provider.py

```
@@ -391,6 +391,69 @@ async def mock_safe_request(**kwargs):
+        assert "generationConfig" in captured_payload
+        gen_config = captured_payload["generationConfig"]
+
+        # CRITICAL: Structured output should be DISABLED when URLs are present
+        assert "responseMimeType" not in gen_config, \
+            "URL context incompatible with structured output - responseMimeType should be disabled"
+        assert "responseJsonSchema" not in gen_config, \
+            "URL context incompatible with structured output - responseJsonSchema should be disabled"
+
+        # Verify URL tool is still present
+        assert "tools" in captured_payload
+        tool_names = [tool.get("google_search") for tool in captured_payload["tools"] if "google_search" in tool]
```

Ruff (0.14.5)

Local variable `tool_names` is assigned to but never used

Remove assignment to unused variable `tool_names`

(F841)